src/Juvix/Compiler/Pipeline/Lockfile.hs

```diff
@@ -0,0 +1,134 @@
+    where
+      dep :: (Key, Value)
+      dep = case i ^. lockfileDependencyDependency of
+        DependencyGit g -> (Str.git, toJSON g)
+        DependencyPath p -> (Str.path_, toJSON p)
+  toEncoding i = pairs (dep <> pair Str.dependencies (toEncoding (i ^. lockfileDependencyDependencies)))
+    where
+      dep :: Series
+      dep = case i ^. lockfileDependencyDependency of
+        DependencyGit g -> pair Str.git (toEncoding g)
+        DependencyPath p -> pair Str.path_ (toEncoding p)
+
+instance FromJSON LockfileDependency where
+  parseJSON = toAesonParser' p
+    where
+      p :: Parse' LockfileDependency
+      p = do
+        _lockfileDependencyDependency <- p'
+        _lockfileDependencyDependencies <- key Str.dependencies fromAesonParser
+        return LockfileDependency {..}
+
+      p' :: Parse' Dependency
+      p' = DependencyPath <$> (key Str.path_ fromAesonParser) Aeson.<|> DependencyGit <$> (key Str.git fromAesonParser)
+
+lockfileOptions :: Options
+lockfileOptions =
+  defaultOptions
+    { fieldLabelModifier = over Lens._head toLower . dropPrefix "_lockfile",
+      rejectUnknownFields = True,
+      omitNothingFields = True
+    }
+
+instance ToJSON Lockfile where
+  toJSON = genericToJSON lockfileOptions
+  toEncoding = genericToEncoding lockfileOptions
+
+instance FromJSON Lockfile where
+  parseJSON = toAesonParser' (Lockfile <$> (key Str.dependencies fromAesonParser))
+
+mkPackageLockfilePath :: Path Abs Dir -> Path Abs File
+mkPackageLockfilePath = (<//> juvixLockfile)
+
+mayReadLockfile ::
+  forall r.
+  (Members '[Files, Error Text] r) =>
+  Path Abs Dir ->
+  Sem r (Maybe LockfileInfo)
+mayReadLockfile root = do
+  let lockfilePath = mkPackageLockfilePath root
+  lockfileExists <- fileExists' lockfilePath
+  if
+      | lockfileExists -> do
+          bs <- readFileBS' lockfilePath
+          either (throw . pack . prettyPrintParseException) ((return . Just) . mkLockfileInfo lockfilePath) (decodeEither' @Lockfile bs)
+      | otherwise -> return Nothing
+  where
+    mkLockfileInfo :: Path Abs File -> Lockfile -> LockfileInfo
+    mkLockfileInfo _lockfileInfoPath _lockfileInfoLockfile = LockfileInfo {..}
+
+lockfileEncodeConfig :: Config
+lockfileEncodeConfig = setConfCompare keyCompare defConfig
+  where
+    -- serialize the dependencies after all other keys
+    keyCompare :: Text -> Text -> Ordering
+    keyCompare x y =
+      if
+          | y == Str.dependencies || x == Str.dependencies -> GT
+          | otherwise -> compare x y
+
+writeLockfile :: (Members '[Files] r) => Path Abs Dir -> Lockfile -> Sem r ()
+writeLockfile root lf = do
+  ensureDir' (parent lockfilePath)
+  writeFileBS lockfilePath (header <> encodePretty lockfileEncodeConfig lf)
+  where
+    lockfilePath :: Path Abs File
+    lockfilePath = mkPackageLockfilePath root
+
+    header :: ByteString
+    header = "# This file was autogenerated by Juvix.\n# Do not edit this file manually.\n\n"
```

Comment:
Fix this message to be:

```
# File automatically generated by Juvix version X.X.X
```

The other line is fine.

Reply:
done